Userland/Libraries/LibWeb/Fetch/Infrastructure/HTTP/Requests.cpp

```diff
@@ -265,6 +266,66 @@ ErrorOr<void> Request::add_range_header(u64 first, Optional<u64> const& last)
     return {};
 }
 
+// https://fetch.spec.whatwg.org/#append-a-request-origin-header
+ErrorOr<void> Request::add_origin_header()
+{
+    // 1. Let serializedOrigin be the result of byte-serializing a request origin with request.
+    auto serialized_origin = TRY(byte_serialize_origin());
+
+    // 2. If request’s response tainting is "cors" or request’s mode is "websocket", then append (`Origin`, serializedOrigin) to request’s header list.
+    if (m_response_tainting == ResponseTainting::CORS || m_mode == Mode::WebSocket) {
+        auto header = Header {
+            .name = MUST(ByteBuffer::copy("Origin"sv.bytes())),
```

Comment:
We're in an error or context here, why must and not try?

Reply:
https://github.com/SerenityOS/serenity/commit/02a4cba0860bfb85da5dc3823f585c42ac94eb89